pkg/didcomm/protocol/didexchange/service.go

```diff
@@ -299,47 +296,97 @@ func (s *Service) sendMsgEvents(msg *service.StateMsg) {
 func (s *Service) startInternalListener() {
 	go func() {
 		for msg := range s.callbackChannel {
-			// TODO handle error in callback - https://github.com/hyperledger/aries-framework-go/issues/242
-			if err := s.process(msg); err != nil {
-				// TODO handle error
-				logger.Errorf(err.Error())
+			if msg.Err != nil {
+				// TODO https://github.com/hyperledger/aries-framework-go/issues/438 Cleanup/Update data on Stop Event Action
+				logger.Errorf("client action event processing failed - msgID:%s error:%s", msg.ID, msg.Err)
+				err := s.processCallbackFailure(msg.ID, msg.Err)
```

Comment:
Suggest we rename this to "abandon" or something similar

Reply:
Any specific suggestions ?

Reply:
Missed the point. Updated the [name](https://github.com/hyperledger/aries-framework-go/compare/b3d55db4c56ee239eef3f187fef8efba3642ff2b..316bca19317a19ab632e20dc1d3f2dca858ebbb3#diff-f761c91e753eba1c42a68076df06ce12R334).